src/HttpClient.php

```diff
@@ -90,7 +136,10 @@ public function deploy(array $data)
 
         $data['apiKey'] = $this->config->getApiKey();
 
-        $this->post('deploy', ['json' => $data]);
+        $this->post(
+            $this->config->getNotifyEndpoint().'deploy',
```

Comment:
This change is quite fragile as it relies on the notify endpoint ending with a `/`, which we don't guarantee

I'm not sure this method is worth keeping though — it's been deprecated since https://github.com/bugsnag/bugsnag-php/pull/454 more than two years ago. We don't call this internally and the `HttpClient` isn't exposed anywhere. That means the only way to call this `deploy` method is by reflection trickery or extending `Client` and calling it from a custom implementation

Having said that, adding a `/` if one doesn't exist isn't difficult so we can also keep this around if that's preferable

Reply:
I agree, we should look to remove if this ends up in a major release.

However, let's stick on a trivial check for a slash for now just so we maintain robust BC for now.